hpy/devel/include/hpy.h

```diff
@@ -140,6 +140,7 @@ typedef struct _HPyContext_s HPyContext;
 #include "hpy/runtime/helpers.h"
 
 #ifdef HPY_UNIVERSAL_ABI
+typedef struct { void* _data; } HPyThreadState;
```

Comment:
I think I went to far here, we should have this type defined the same for both CPython/Universal (here I have different typedef for CPython ABI aliasing this directly to `PyThreadState`) I'll fix that, but it otherwise does not change much from higher level view on this PR.